compiler/src/dotty/tools/dotc/transform/ExplicitOuter.scala

```diff
@@ -155,9 +155,28 @@ object ExplicitOuter {
   private def newOuterAccessors(cls: ClassSymbol)(implicit ctx: Context) =
     newOuterAccessor(cls, cls) :: (if (cls is Trait) Nil else newOuterParamAccessor(cls) :: Nil)
 
-  /** A new outer accessor or param accessor */
+  /** A new outer accessor or param accessor.
+   *  @param  owner  The class where the outer accessor is located
+   *  @param  cls    The class relative to which the outer is computed (can be a base class of owner)
+   *  @param  name   The name of the outer access
+   *  @param  flags  The flags of the outer accessor
+   *
```

Comment:
Complication: outer references now need to have correct type arguments where previously a TypeRef was enough.

Actually, this likely fixes errors that went undetected before, because we should have updated the refinements of outer references but didn't (casts can hide such errors all too conveniently).